### PR TITLE
pkg/trace/agent: ensure normalizer reports bad service name

### PR DESCRIPTION
**Pull request by gbbr**
This change ensure that the normalizer reports a bad service name.
Previously it would always report the empty string on error.

### CODE REVIEW

**Review by ufoot — DISMISSED** [review]
Yup this is better, as a side note I think resource has a similar behavior:

https://github.com/DataDog/datadog-agent/pull/2942/files#diff-4140e394b1bfca821a710b030200a141R64

**Comment by gbbr** [comment]
@ufoot you're right. I've made that change. Please re-approve.

File: pkg/trace/agent/normalizer.go [hunk 1]
```diff
@@ -94,7 +96,11 @@ func Normalize(s *pb.Span) error {
 	// ParentID set on the client side, no way of checking
 
 	// Type
-	s.Type = toUTF8(s.Type)
+	typ := toUTF8(s.Type)
```
Comment (ufoot) [inline pkg/trace/agent/normalizer.go]: good catch !

**Review by ufoot — APPROVED** [review]
Thanks for making this more reliable.

**Pull request merged**